src/ssl.c

```diff
@@ -17772,16 +17777,42 @@ size_t wolfSSL_get_client_random(const WOLFSSL* ssl, unsigned char* out,
 #endif
     }
 
+    /* If a valid struct is provided with function pointers, will override
+       RAND_seed, bytes, cleanup, add, pseudo_bytes and status.
+
+       Returns SSL_SUCCESS on success, SSL_FAILURE on failure. */
+    int wolfSSL_RAND_set_rand_method(const WOLFSSL_RAND_METHOD *methods)
+    {
+        if (methods == NULL) {
+            return SSL_FAILURE;
+        }
+
+        if (rand_methods == NULL) {
```

Comment:
This function can just use the incoming const methods pointer directly. At least that's the assumption openssl makes. Then you can avoid a malloc / copy and free.

There is no locking on the use of `rand_methods` and I think there should be for multi-threaded cases. Let me know if you need some tips for adding the mutex's around the use of `rand_methods`

Reply:
Updated to use the pointer directly, and added a mutex which locks on all writes/reads to rand_methods.